audio/decoders/wave.cpp

```diff
@@ -62,15 +75,22 @@ bool loadWAVFromStream(Common::SeekableReadStream &stream, int &size, int &rate,
 		stream.read(buf, 4);
 	}
 
-	if (memcmp(buf, "JUNK", 4) == 0) {
-		uint32 junksize = stream.readUint32LE();
-		// skip junk padding (add 1 byte if odd)
-		stream.skip(junksize + (junksize % 2));
-		stream.read(buf, 4);
+	while (1) { // skip junk/bext... chunks
+		int i;
+		for (i = 0; (i < EXT_CHUNKS) && (memcmp(buf, extensionChunks[i], 4) != 0); i++)
+			;
+		if (i != EXT_CHUNKS) { // found a known chunk
+			uint32 chunkSize = stream.readUint32LE();
+			// skip junk/ext chunk (add 1 byte if odd)
+			stream.skip(chunkSize + (chunkSize % 2));
```

Comment:
Does this apply to all these chunks?
There's code above for skipping "fact" chunks, that doesn't have this rounding up. Should it also be included with these chunks?

Reply:
The chunk size should always be even, it's indicated in the specs for the JUNK chunk https://www.daubnet.com/en/file-format-riff; I couldn't find mention of this among the BWF specs though.
I used this tensorflow implementation as a reference and it does the rounding up for all chunks too
https://github.com/tensorflow/io/blob/189e5e1205d03b84d1b44c2e97cf9e97c5f910be/tensorflow_io/core/kernels/audio_video_wav_kernels.cc#L98

Reply:
Thanks for the reference and the explanation :)